app/Http/Controllers/Api/V1/PlatformConnectionController.php

```diff
@@ -24,69 +25,64 @@ public function index(): \Illuminate\Http\Resources\Json\AnonymousResourceCollec
         return  PlatformConnectionResource::collection(PlatformConnection::all());
     }
 
+    // look up a relative connection per consumer platform id
+    // *************************************************************
+    // GET: /api/v1/consumer_platforms/{id}/platform_connection
+    // Request $req example:
+    //  "consumer_id": 1 || "provider_id": 1
+    // *************************************************************
+    public function show(string $id): PlatformConnectionResource
+    {
+        $platform = PlatformConnection::where('consumer_platform_id', $id)->first();
+        if (!$platform) {
+            return response()->json(['error' => 'No matching platform connection found'], 401);
+        }
+        return new PlatformConnectionResource($platform);
```

Comment:
@PThorpe92 in order to be able to return the requisite Provider Platform data (and Connection state) to the v.1 UI this needs to return the `ConsumerProviderResource::collection` that was in the https://github.com/UnlockedLabs/starfish/pull/3 request that you had integrated into your branch earlier when we collaborated.

Reply:
ConsumerProviderResource?

The `show` function you have in the link returns 
```php
public function show(ConsumerPlatform $consumerPlatform)
    {
        return ConsumerPlatformResource::make($consumerPlatform);
    }
```

So because this is a `show` function and you have specified the `ID` in the URL, it should return a collection of all that consumer platforms connections? that does make sense.

Reply:
Yes. Additionally it gets the `state` value of the Platform Connection and adds that to the json Provider Platform (for that particular Consumer Platform) response data